beater/handlers.go

```diff
@@ -133,6 +137,60 @@ func isAuthorized(req *http.Request, secretToken string) bool {
 	return subtle.ConstantTimeCompare([]byte(parts[1]), []byte(secretToken)) == 1
 }
 
+func corsHandler(allowedOrigin string, h http.Handler) http.Handler {
+
+	var isSupported = func(requested string, allowed string) bool {
+		rs := true
+		for _, s := range strings.Split(requested, ",") {
+			if !strings.Contains(allowed, s) {
+				rs = false
+			}
+		}
+		return rs
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		// origin header is always set by the browser
+		origin := r.Header.Get("Origin")
+		validOrigin := allowedOrigin == "*" || origin == allowedOrigin
+
+		if r.Method == "OPTIONS" {
+
+			requestHeaders := r.Header.Get("Access-Control-Request-Headers")
```

Comment:
For the OPTIONS request, it should be enough to simply return a static list of allowed headers and methods. We don't need to actually worry about what headers or methods the browser is asking for.

As long as the OPTIONS response includes what it needs (e.g. `Request-Method` is `POST` and the answer is `Allow-Methods: POST, OPTIONS`) it will allow the POST.

As far as i can tell, this means the logic for splitting `requested` goes away.